pkg/controller/security/manager_test.go

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2024 The Kmesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package security
+
+import (
+	"testing"
+	"time"
+
+	"github.com/agiledragon/gomonkey/v2"
+	"github.com/stretchr/testify/assert"
+	"istio.io/istio/pkg/security"
+	camock "kmesh.net/kmesh/pkg/controller/security/mock"
+)
+
+// refCnt test
+func TestFileSecrets(t *testing.T) {
+
+	t.Run("TestrefCnt", func(t *testing.T) {
+		runTestRefCnt(t)
+	})
+	t.Run("Testrotate", func(t *testing.T) {
+		runTestRotate(t)
+	})
+}
+
+func runTestRefCnt(t *testing.T) {
+	patches := gomonkey.NewPatches()
+	patches.ApplyFunc(newCaClient, func(opts *security.Options, tlsOpts *tlsOptions) (Client, error){
+		return camock.NewMockCaClient(opts, 2 * time.Hour)
+	})
+	defer func() {
+		patches.Reset()
+	}()
+
+	secretManager, err := NewSecretManager()
+	if err != nil {
+		t.Errorf("NewSecretManager failed %v", err)
+	} 
+
+	identity1 := "identity1"
+	identity2 := "identity2"
+	
+	secretManager.SendCertRequest(identity1, ADD)
+	secretManager.SendCertRequest(identity1, ADD)
+	secretManager.SendCertRequest(identity2, ADD)
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Here's why needs a sleep

Reply:
The subsequent processing is performed in the routine and takes a certain period of time. To ensure the stability of the test case, sleep

Reply:
can you add a notify mechanism in the async go routine